core/src/exceptions.ts

```diff
@@ -94,124 +85,186 @@ export abstract class GardenBaseError<D extends object = any> extends Error impl
+ * - A Javascript TypeError has occurred, e.g. reading property on undefined.
+ * - "This should not happen" kind of situations, e.g. internal data structures are in an invalid state.
+ * - An unhandled exception has been thrown by a library. If you don't know what to do with this exception and it is most likely not due to user error, wrap it with "InternalError".
+ *
+ * In case the network is involved, we should *not* use the "InternalError", because that's usually a situation that the user needs to resolve.
+ */
+export class InternalError extends GardenError {
+  // we want it to be obvious in amplitude data that this is not a normal error condition
+  type = "crash"
+
+  // not using object destructuring here on purpose, because errors are of type any and then the error might be passed as the params object accidentally.
+  static wrapError(error: Error | string | any, detail?: unknown, prefix?: string): InternalError {
+    let message: string | undefined
+    let stack: string | undefined
+
+    if (error instanceof Error) {
+      message = error.message
+      stack = error.stack
+    } else if (isString(error)) {
+      message = error
+    } else if (error) {
+      message = error["message"]
+      stack = error["stack"]
+    }
 
-  constructor(error: Error) {
-    super({ message: error.message, stack: error.stack })
+    message = message ? stripAnsi(message) : ""
+
+    return new InternalError({ message: prefix ? `${stripAnsi(prefix)}: ${message}` : message, stack, detail })
   }
 }
 
-export function toGardenError(err: Error | GardenBaseError | string): GardenBaseError {
-  if (err instanceof GardenBaseError) {
+export function toGardenError(err: Error | GardenError | string | any): GardenError {
+  if (err instanceof GardenError) {
     return err
-  } else if (err instanceof Error) {
-    const wrappedError = new WrappedNativeError(err)
-    const out = new RuntimeError({ message: err.message, wrappedErrors: [wrappedError] })
-    out.stack = err.stack
-    return out
-  } else if (isString(err)) {
-    return new RuntimeError({ message: err })
   } else {
-    const msg = err["message"]
-    return new RuntimeError({ message: msg })
+    return InternalError.wrapError(err)
   }
 }
 
-function filterErrorDetail(detail: any) {
-  return withoutInternalFields(sanitizeValue(detail))
+export function explainGardenError(rawError: GardenError | Error | string, context?: string) {
+  const error = toGardenError(rawError)
+
+  let errorMessage = error.message.trim()
+
+  // If this is an unexpected error, we want to output more details by default and provide some guidance for the user.
+  if (error instanceof InternalError) {
+    let bugReportInformation = formatGardenErrorWithDetail(error)
+
+    if (context) {
+      bugReportInformation = `${stripAnsi(context)}\n${bugReportInformation}`
+    }
+
+    return chalk.red(dedent`
+    ${chalk.bold("Encountered an unexpected Garden error. We are sorry for this. This is likely a bug 🍂")}
```

Comment:
```suggestion
    ${chalk.bold("Encountered an unexpected Garden error. This is likely a bug 🍂")}
```
I appreciate the tone, but I personally think the apology is maybe unnecessary, plus "we" is sort of undefined for an open-source project :) If we want to include one, we should at least change the phrasing slightly. _Maybe_ "Sorry about that!" but it's also fine imo to just skip.

Reply:
OK! O felt it was necessary because the user cannot do anything about these errors and you can feel quite helpless in such a situation. I removed it because it doesn't really help with the helplessness anyway.